tests/e2e/mock_llm.py

```
@@ -331,73 +337,190 @@ def _next_llm_trace_response(
+def _trace_tool_results_with_recorded_ids(
+    state: dict,
+    messages: list[dict],
+) -> list[dict]:
+    """Present runtime results under their stable fixture IDs."""
+    runtime_to_recorded = {
+        runtime_id: recorded_id
+        for recorded_id, runtime_id in state["tool_call_id_aliases"].items()
     }
+    results = []
+    for result in _find_tool_results(messages):
+        result = dict(result)
```

Ruff (0.15.21)

[warning] `for` loop variable `result` overwritten by assignment target

(PLW2901)